DEVX-1130: Add C3 to Multi Region Replication demo

awalther28 opened the pull request:
### Description 

https://confluentinc.atlassian.net/browse/DEVX-1130

_What behavior does this PR change, and why?_

Adds C3 to the MRC demo + augments tutorial steps with C3 screenshots. 

### Author Validation

_Describe the validation already done, or needs to be done, by the PR submitter._

<!-- Uncomment any of the following that are required -->
 - [x] Documentation 
<!-- - [ ] ccloud -->
<!-- - [ ] ccloud/beginner-cloud -->
<!-- - [ ] ccloud/ccloud-stack -->
<!-- - [ ] clickstream -->
<!-- - [ ] clients/avro -->
<!-- - [ ] clients/cloud -->
<!-- - [ ] cloud-etl -->
<!-- - [ ] connect-streams-pipeline -->
<!-- - [ ] cp-quickstart -->
<!-- - [ ] kubernetes/gke-base -->
<!-- - [ ] kubernetes/replicator-gke-cc -->
<!-- - [ ] microservices-orders -->
<!-- - [ ] multi-datacenter -->
- [x] multiregion 
<!-- - [ ] music -->
<!-- - [ ] replicator-schema-translation -->
<!-- - [ ] replicator-security -->
<!-- - [ ] security/rbac -->
<!-- - [ ] security/secret-protection -->


### Reviewer Tasks

_Describe the tasks/validation that the PR submitter is requesting to be done by the reviewer._
Docs link: http://staging-docs-independent.confluent.io.s3-website-us-west-2.amazonaws.com/docs-platform/PR/208/6.1.0-SNAPSHOT/tutorials/examples/multiregion/docs/multiregion.html#
<!-- Uncomment any of the following that are required -->
 - [ ] Documentation 
 - [ ] multiregion
<!-- - [ ] ccloud -->
<!-- - [ ] ccloud/beginner-cloud -->
<!-- - [ ] ccloud/ccloud-stack -->
<!-- - [ ] clickstream -->
<!-- - [ ] clients/avro -->
<!-- - [ ] clients/cloud -->
<!-- - [ ] cloud-etl -->
<!-- - [ ] connect-streams-pipeline -->
<!-- - [ ] cp-quickstart -->
<!-- - [ ] kubernetes/gke-base -->
<!-- - [ ] kubernetes/replicator-gke-cc -->
<!-- - [ ] microservices-orders -->
<!-- - [ ] multi-datacenter -->

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
DEVX-1130; add c3

awalther28 committed:
Merge branch 'DEVX-1130' of github.com:confluentinc/examples into DEVX-1130

awalther28 committed:
DEVX-1130; swap to metrics cluster

awalther28 committed:
DEVX-1130; swap to metrics cluster

awalther28 committed:
DEVX-1130; swap to metrics cluster

awalther28 committed:
DEVX-1130; swap to metrics cluster

ybyzek committed:
Enhancements to DEVX-1130 (MRC + C3) (#858)

* Metrics cluster: 1 instead of 2 brokers

* Tweak

ybyzek committed:
Enhancements to DEVX-1130 (MRC + C3) (#858)

ybyzek committed:
Enhancements to DEVX-1130 (MRC + C3) (#858)

* Metrics cluster: 1 instead of 2 brokers

* Tweak

ybyzek committed:
Enhancements to DEVX-1130 (MRC + C3) (#858)

* Metrics cluster: 1 instead of 2 brokers

* Tweak

awalther28 committed:
DEVX-1130; start adding C3 screenshots

awalther28 committed:
DEVX-1130; start adding C3 screenshots

awalther28 committed:
DEVX-1130; start adding C3 screenshots

awalther28 committed:
DEVX-1130; start adding C3 screenshots

awalther28 committed:
DEVX-1130; add space

awalther28 committed:
DEVX-1130; add space

awalther28 committed:
DEVX-1130; add space

awalther28 committed:
DEVX-1130; add screenshots through almost end

awalther28 committed:
DEVX-1130; add screenshots through almost end

awalther28 committed:
DEVX-1130; add screenshots through almost end

awalther28 committed:
DEVX-1130; fix merge

awalther28 committed:
DEVX-1130: remove ====

xli1996 committed:
Bump Confluent to 6.2.0-SNAPSHOT, Kafka to 6.2.0-SNAPSHOT

xli1996 committed:
Bump Confluent to 6.2.0-SNAPSHOT, Kafka to 6.2.0-SNAPSHOT

xli1996 committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

awalther28 committed:
DEVX-1130: update to reflect automatic observer promotion

awalther28 committed:
DEVX-1130: update to reflect automatic observer promotion

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

awalther28 committed:
Merge branch '6.1.x' into master

ConfluentJenkins committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

awalther28 committed:
DEVX-1130: fix screenshots

awalther28 committed:
DEVX-1130: fix screenshots

awalther28 committed:
DEVX-1130: remove non used pngs

awalther28 committed:
DEVX-1130: remove non used pngs

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

rspurgeon committed:
Merge branch '6.1.x'

ybyzek committed:
Merge branch '6.1.x'

awalther28 committed:
DEVX-1130: fix screenshots

awalther28 committed:
DEVX-1130: fix screenshots

awalther28 committed:
DEVX-1130:fix file name

awalther28 committed:
DEVX-1130:fix file name

ybyzek committed:
Merge branch '6.1.x'

awalther28 committed:
Merge remote-tracking branch 'origin/master' into DEVX-1130

ybyzek commented:
@awalther28 

1. Woohoo! 🎆 
2. This PR should target `6.1.x`, not `master`
3. Can you add some verbiage at the top of the docs page that mentions C3?  Otherwise, first mention of C3 is in a step that uses it.  Potential mentions: `Overview` ("hey, and you can monitor your cluster in C3...") and `Configuration`

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -378,12 +381,25 @@ You could create all the topics by running the script :devx-examples:`create-top
+
+   |C3 cluster navigation|
+
+   Click on the "mrc" cluster, then make your way to the "Topics" section.
```
It would be helpful to explicitly explain the difference between the 2 clusters

awalther28 replied:
the latest docs preview link reflects an explanation of the 2 clusters

ybyzek replied:
Awesome, thanks!

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -556,6 +572,9 @@ There is a script you can run to collect the JMX metrics from the command line,
       multi-region-async-op-leader-is-observer: 0
       multi-region-default: 0
 
+#. Metrics relating to partitions, followers and observers are viewable in the "Topics" section of the C3 UI.
```
C3 doesn't show exactly the same JMX metrics as the output above.  Is the screenshot needed at all if this is the same diagram as shown earlier?

awalther28 replied:
It has been removed and the latest docs preview link reflects that.

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -746,6 +768,9 @@ In this section, you will simulate a region failure by bringing down the ``west`
       Topic: multi-region-default PartitionCount: 1   ReplicationFactor: 4    Configs: min.insync.replicas=1,confluent.placement.constraints={"version":1,"replicas":[{"count":2,"constraints":{"rack":"west"}}],"observers":[{"count":2,"constraints":{"rack":"east"}}]}
          Topic: multi-region-default Partition: 0    Leader: none    Replicas: 2,1,3,4   Isr: 1  Offline: 2,1    Observers: 3,4
 
+#. Given five minutes, you should see something similar to below in C3. The numbers may be off while the cluster stabilizes.
```
```suggestion
#. After the |c3| cluster metrics stabilize in about five minutes, you should see output similar to below.
```

awalther28 committed:
Update multiregion/docs/multiregion.rst

Co-authored-by: Yeva Byzek <ybyzek@users.noreply.github.com>

awalther28 committed:
DEVX-1130: address comments

awalther28 committed:
DEVX-1130: address comments

awalther28 committed:
DEVX-1130: shorten C3

awalther28 commented:
@ybyzek Thanks for reviewing on your day off 😉  I believe I have addressed all of your comments in the latest commits. The docs preview link should reflect the latest changes.

ybyzek commented on utils/config.env:
```diff
@@ -5,13 +5,13 @@
-CONFLUENT=6.1.0-0
-CONFLUENT_DOCKER_TAG=6.1.x-latest
-CONFLUENT_SHORT=6.1
+CONFLUENT=6.2.0-0
```
@awalther28 I think this change might have been mistakenly brought in by the retargeting (maybe the branch needs a rebase?) IIUC this file was not touched by this PR, so this needs to be reverted.  This applies to a few files:

- utils/config.env
- security/README.md
- replicator-schema-translation/docs/conf.py
- microservices-orders/docs/conf.py
- README.md
- ccloud/docs/config/*
- a few others...

andrewegel committed:
Set Confluent to 5.5.3, Kafka to 5.5.3.

andrewegel committed:
Merge branch '5.5.3-post' into 6.0.0-post

andrewegel committed:
Merge branch '6.0.0-post' into 6.0.1-post

andrewegel committed:
Merge branch '5.5.3-post' into 5.5.x

andrewegel committed:
Merge branch '6.0.1-post' into 6.0.x

andrewegel committed:
Merge branch '5.5.x' into 6.0.x

andrewegel committed:
Merge branch '6.0.x' into 6.1.x

ybyzek committed:
DEVX-2361: broken link in client examples: to Schema Management in CCloud (#889)

ybyzek committed:
Merge branch '6.0.1-post' into 6.0.x

ybyzek committed:
Merge branch '6.0.x' into 6.1.x

awalther28 committed:
Merge branch 'DEVX-1130' of github.com:confluentinc/examples into DEVX-1130

awalther28 committed:
DEVX-1130: revert modified files back to origin/6.1.x

awalther28 commented:
Great catch on the rebase files @ybyzek. I resolved all but one file that I can't seem to get resolved. I've been using `git checkout origin/6.1.x -- clients/docs/includes/client-example-schema-registry-1.rst` with the latest branches pulled locally yet no luck. Any thoughts on other methods I could use here?

ybyzek commented:
@awalther28 I forgot the issue you’re encountering, but is this the file you’re looking for and can it be manually added in?

https://github.com/confluentinc/examples/blob/6.0.1-post/clients/docs/includes/client-example-schema-registry-1.rst

awalther28 committed:
DEVX-1130: attempt to manually revert file

awalther28 committed:
DEVX-1130: attempt to manually revert file

awalther28 commented:
Okay manually added, @ybyzek this PR not solely reflects the multiregion changes. Do you have other comments on how I can improve this PR?

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -378,12 +385,25 @@ You could create all the topics by running the script :devx-examples:`create-top
+
+   |C3 cluster navigation|
+
+   You will notice 2 clusters, "mrc" which is the multiregion cluster and "metrics" which is a dedicated metrics cluster that runs Control Center. By hosting Control Center on its own Kafka cluster, it is independent of the availability of the production cluster it is monitoring. We will be working on topics within the "mrc" cluster for the remainder of this tutorial. Click on the "mrc" cluster,  ...
```
```suggestion
   Notice two clusters: "mrc" which is the multiregion cluster, and "metrics" which is a dedicated metrics cluster that runs |c3|. By backing |c3| to its own |ak| cluster, it has no dependency on the availability of the production cluster it is monitoring. The remainder of this tutorial works on topics within the "mrc" cluster. Click on the "mrc" cluster, then make your way to the "Topics" section.
```

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -13,6 +13,8 @@ Often referred to as a stretch cluster, |mrrep| replicate data between datacente
 You can choose how to replicate data, synchronously or asynchronously, on a per |ak| topic basis.
 It provides good durability guarantees and makes disaster recovery (DR) much easier.
 
+We will use a dedicated metrics cluster running |c3| to monitor the Multi-Region Cluster.
```
```suggestion
This tutorial uses a dedicated |ak| cluster backing |c3|, which monitors the multi-region cluster.
```

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -378,12 +385,25 @@ You could create all the topics by running the script :devx-examples:`create-top
+
+   |C3 topics overview|
+
+   Click on each topic to see details about the replica and observer placement. This will mirror the CLI output above. Below is an example of the multi-region-async topic.
```
```suggestion
   Click on each topic to see details about the replica and observer placement. |c3| matches the CLI output above. Below is an example of the ``multi-region-async`` topic.
```

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -1113,6 +1147,8 @@ Now you will bring region ``west`` back online and restore configuration to the
      region is restored. Leader election is not required for this demotion
      process, it will happen as soon as the failed region is restored.
 
+   - The C3 topics page is the same as it was at the start of this tutorial.
```
```suggestion
   - The |c3| topics page is the same as it was at the start of this tutorial.
```

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -1243,23 +1279,51 @@ it is possible Docker networking not working or cleaning up properly between run
+
+.. |multi-region-async replicas|
+image:: images/multi-region-async-placement.png
+   :alt: multi-region-async replicas
```
@awalther28 WDYT about adding the following to make the graphic smaller?

```
      :width: 500px
```

awalther28 replied:
The smaller picture looks much better. 
<img width="1108" alt="Screen Shot 2021-01-04 at 11 56 39 AM" src="https://user-images.githubusercontent.com/12937975/103569093-eedbda00-4e83-11eb-84d4-ad78f22732f7.png">

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -378,12 +385,25 @@ You could create all the topics by running the script :devx-examples:`create-top
    - The ``multi-region-async``, ``multi-region-async-op-under-min-isr``, ``multi-region-async-op-under-replicated``, ``multi-region-async-op-leader-is-observer`` and ``multi-region-default`` topics have replicas
      across ``west`` and ``east`` regions, but only 1 and 2 are in the ISR, and 3 and
-     4 are observers.
+     4 are observers. This can be observed via the CLI output or C3.
```
```suggestion
     4 are observers. This can be observed via the CLI output or |c3|.
```

ybyzek commented on multiregion/docs/multiregion.rst:
```diff
@@ -613,6 +633,9 @@ In this section, you will simulate a single broker failure in the ``west`` regio
       Topic: multi-region-default	PartitionCount: 1	ReplicationFactor: 4	Configs: min.insync.replicas=1,confluent.placement.constraints={"version":1,"replicas":[{"count":2,"constraints":{"rack":"west"}}],"observers":[{"count":2,"constraints":{"rack":"east"}}]}
       	Topic: multi-region-default	Partition: 0	Leader: 2	Replicas: 1,2,3,4	Isr: 2	Offline: 1	Observers: 3,4
 
+#. Verify similar replica placement in C3. Note that it may take up to 5 minutes for C3 to properly report the new topic stats.
```
```suggestion
#. Verify similar replica placement in |c3|. Note that it may take up to 5 minutes it to properly report the new topic stats.
```

ybyzek replied:
@awalther28 note that `C3` is an internal term, but we should use `Control Center` or `Confluent Control Center` in documentation.  

```
.. |c3-short| replace:: Control Center
.. |c3| replace:: Confluent Control Center
```

ybyzek reviewed (approved):
@awalther28 I left some documentation tweaks, and after those are addressed, LGTM.  I suggest we not merge this PR yet, let's give the other reviewers (@thomaskwscott , @lukeknep ) a day to optionally review.

awalther28 committed:
Apply suggestions from code review

Co-authored-by: Yeva Byzek <ybyzek@users.noreply.github.com>

awalther28 committed:
DEVX-1130: remove C3 references + make image smaller

awalther28 committed:
DEVX-1130: fix image alt text

awalther28 committed:
DEVX-1130: attempt to fix rst images

awalther28 committed:
DEVX-1130: attempt to fix rst images
